include/devicetree.h

```diff
@@ -900,6 +900,26 @@
  */
 #define DT_HAS_NODE_STATUS_OKAY(node_id) IS_ENABLED(DT_CAT(node_id, _EXISTS))
 
+/**
+ * @brief Does a node label refer to a usable node?
+ *
+ * Example uses:
+ *
+ *     DT_HAS_NODELABEL_STATUS_OKAY(a_nodelabel)
+ *
+ * Tests whether a node label refers to a node which:
+ *
+ * - exists in the devicetree, and
+ * - is enabled (has status property "okay" or no status property), and
```

Comment:
This documentation is inconsistent with `DT_HAS_NODE_STATUS_OKAY()` which only says it's enabled if it has status property "okay".

Why does this need to be extended to support enabled when there is no `status` property?

Reply:
Because that is how 'enabled' has always been "defined".   We need to update the docs for the other macro.  A missing 'status' is treated the same as 'status' = 'okay'.

Reply:
OK.  It might be best to just say "is enabled", and leave what that means unspecified at this level..  Now it's "status = okay or not defined", in the future it will be something else, and it would be nice to avoid having to add new API once that something else is implemented.

Reply:
> A missing 'status' is treated the same as 'status' = 'okay'.

That's true currently, but is it really what we want going forward? In https://github.com/zephyrproject-rtos/zephyr/pull/25009 I've gone another way where explicit 'okay' is required, and other than having to add it to a few sensors in the DT test app.overlay, it hasn't been the end of the world.

Arguably the device should need to be explicitly enabled, no? I can't find anything in dtspec arguing that status has a default value; is that common practice in Linux to assume it's "okay"? I can change 25009 if so.

Reply:
Its common practice in linux that 'okay' missing means 'okay'.

Reply:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/of/base.c#n581